pkg/operator/controller/ingress/controller.go

```diff
@@ -389,10 +389,20 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig
 	// Detect changes to GCP LB provider parameters, which is something we can safely roll out.
 	statusLB := ic.Status.EndpointPublishingStrategy.LoadBalancer
 	specLB := effectiveStrategy.LoadBalancer
-	if specLB != nil && statusLB != nil && specLB.ProviderParameters != nil && statusLB.ProviderParameters != nil &&
-		specLB.ProviderParameters.GCP != statusLB.ProviderParameters.GCP {
-		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.GCP = effectiveStrategy.LoadBalancer.ProviderParameters.GCP
-		return true
+	if specLB != nil && statusLB != nil {
+		// If the ProviderParameters field does not exist for spec or status,
+		// just propagate (or remove) ProviderParameters in it's entirety.
+		if specLB.ProviderParameters == nil && statusLB.ProviderParameters != nil ||
+			specLB.ProviderParameters != nil && statusLB.ProviderParameters == nil {
```

Comment:
Is there any scenario where `specLB.ProviderParameters` will be nil but `statusLB.ProviderParameters` will be non-nil?

Reply:
Yes, in the case that a user deletes the provider parameters field from spec (which is feasible if they no longer want to specify any GCP Global Access setting).